src/transaction/buffer.rs

```diff
@@ -191,10 +191,28 @@ impl Buffer {
             .await
             .insert(key, BufferEntry::Put(value));
     }
+    
+    /// Mark a value as Insert mutation into the buffer (does not write through).
+    pub async fn insert(&self, key: Key, value: Value) {
+        self.mutations
+            .lock()
+            .await
+            .insert(key, BufferEntry::Insert(value));
```

Comment:
What happens if there exists a mutation of the key when we insert? Should we return an error instead of overwriting it?

Reply:
Done. I add a new pub method to buffer to make it possible for `Transaction` to check whether the key has existed and handle it returning a new kind of error.